src/sage/rings/valuation/inductive_valuation.py

```diff
@@ -0,0 +1,1526 @@
+                    if self.is_gauss_valuation(): # unless in the first step
+                        pass
+                    else:
+                        continue
+
+                verbose("Determining the augmentation of %s for %s"%(self, phi), level=11)
+                old_mu = self(phi)
+                w = self.augmentation(phi, old_mu, check=False)
+
+                # we made some experiments here: instead of computing the
+                # coefficients again from scratch, update the coefficients when
+                # phi - self.phi() is a constant.
+                # It turned out to be slightly slower than just recomputing the
+                # coefficients. The main issue with the approach was that we
+                # needed to keep track of all the coefficients and not just of
+                # the coefficients up to principal_part_bound.
+
+                w_coefficients = w.coefficients(G)
+                if principal_part_bound:
+                    w_coefficients = islice(w_coefficients, 0, principal_part_bound + 1, 1)
+                w_coefficients = list(w_coefficients)
+
+                w_valuations = w.valuations(G, coefficients=w_coefficients)
+                if principal_part_bound:
+                    w_valuations = islice(w_valuations, 0, principal_part_bound + 1, 1)
+                w_valuations = list(w_valuations)
+
+                from sage.geometry.newton_polygon import NewtonPolygon
+                NP = NewtonPolygon(w.newton_polygon(G, valuations=w_valuations).vertices(), last_slope=0)
+
+                verbose("Newton-Polygon for v(phi)=%s : %s"%(self(phi), NP), level=11)
+                slopes = NP.slopes(repetition=True)
+                multiplicities = {slope : len([s for s in slopes if s == slope]) for slope in slopes}
+                slopes = multiplicities.keys()
+                if NP.vertices()[0][0] != 0:
+                    slopes = [-infinity] + slopes
+                    multiplicities[-infinity] = 1
+
+                if not slopes:
+                    q,r = G.quo_rem(phi)
+                    assert not r.is_zero()
+                    phi = phi.coefficients(sparse=False)
+                    for i,c in enumerate(r.coefficients(sparse=False)):
+                        if not c.is_zero():
+                            v = w(c)
+                            # for a correct result we need to add O(pi^v) in degree i
+                            # we try to find the coefficient of phi where such an
+                            # error can be introduced without losing much absolute
+                            # precision on phi
+                            best = i
+                            for j in range(i):
+                                if w(q[j]) < w(q[best]):
+                                    best = j
+                            # now add the right O() to phi in degree i - best
+                            phi[i-best] = phi[i-best].add_bigoh(w(c)-w(q[best]))
```

Comment:
Will this work in ramified extensions?  Your normalization convention is different from that of add_bigoh.  Maybe it's worth grepping for add_bigoh and making sure you use the right normalization everywhere?